src/formatDate.js

```diff
@@ -9,6 +9,74 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+
+    if (element === 'MM') {
+      month = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
+    }
+
+    if (element === 'DD') {
+      day = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
+    }
+
+    if (element === 'YYYY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 5;
+      yearFormat = element;
+    }
+
+    if (element === 'YY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
+      yearFormat = element;
+    }
```

Comment:
The logic for incrementing 'fromCharacter' is flawed. It assumes that there is always a separator after each date part, which might not be the case if the format changes. Instead, increment 'fromCharacter' by the length of the element plus the length of the separator, if there is one.